cmd/cerbosctl/store/export/export.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package export
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	policyv1 "github.com/cerbos/cerbos/api/genpb/cerbos/policy/v1"
+	schemav1 "github.com/cerbos/cerbos/api/genpb/cerbos/schema/v1"
+	"github.com/cerbos/cerbos/client"
+	internalclient "github.com/cerbos/cerbos/cmd/cerbosctl/internal/client"
+	"github.com/cerbos/cerbos/cmd/cerbosctl/store/export/internal"
+	"github.com/cerbos/cerbos/internal/util"
+)
+
+const help = `# Export policies and schemas from the store into a directory
+cerbosctl store export path/to/dir
+
+# Export policies and schemas from the store into a zip archive
+cerbosctl store export archive.zip
+
+# Export policies and schemas from the store into a gzip archive
+cerbosctl store export path/to/archive.gzip
+cerbosctl store export path/to/archive.tar.gz`
+
+type Cmd struct {
+	Path string `arg:"" help:"Path to write policies and schemas" type:"path"`
+}
+
+func (c *Cmd) Run(k *kong.Kong, clientCtx *internalclient.Context) error {
+	policies, err := clientCtx.AdminClient.ListPolicies(context.Background(), client.WithIncludeDisabled())
+	if err != nil {
+		return fmt.Errorf("failed to list policies: %w", err)
+	}
+
+	schemas, err := clientCtx.AdminClient.ListSchemas(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to list schemas: %w", err)
+	}
+
+	exporter, err := internal.NewExporter(c.Path)
+	if err != nil {
+		return fmt.Errorf("failed to create exporter: %w", err)
+	}
+	defer exporter.Close()
+
+	if err := client.BatchAdminClientCall2(context.Background(), clientCtx.AdminClient.GetPolicy, func(ctx context.Context, policies []*policyv1.Policy) error {
+		if policies == nil {
+			return fmt.Errorf("failed to retrieve policy: %w", err)
+		}
+
+		for _, p := range policies {
+			name := strings.ReplaceAll(p.Metadata.StoreIdentifier, "/", "_")
+			if !strings.HasSuffix(name, ".yaml") {
```

Comment:
If a user tries this on a `disk` store that has JSON policies, won't we end up creating a file like `foo.json.yaml`? Since disk-based stores are the only stores that return file names as IDs, I think we should probably honour the actual file type when exporting (JSON -> JSON, YAML -> YAML)